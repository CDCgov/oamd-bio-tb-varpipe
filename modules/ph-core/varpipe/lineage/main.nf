process VARPIPE_LINEAGE {

    tag "${meta.id}"
    label 'process_single'

    container 'oamd-bio-biocontainers-python:3.12_80c558c_v3'

    input:
    tuple val(meta), path(fullFinalAnnotation)
    path(lineages)

    output:
    tuple val(meta), path("*_Lineage.txt"), emit: lineage
    tuple val(meta), path("*.lineage_report.txt"), emit: lineage_report
    path("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'lineage_parser.py'

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_Lineage.txt
    touch ${prefix}.lineage_report.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
    END_VERSIONS
    """
}
