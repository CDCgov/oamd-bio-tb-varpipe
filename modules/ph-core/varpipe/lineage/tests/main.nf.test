nextflow_process {

    name "Test Process VARPIPE_LINEAGE"
    script "../main.nf"
    process "VARPIPE_LINEAGE"

    tag "modules"
    tag "modules_phcore"
    tag "varpipe"
    tag "varpipe/lineage"

    test("varpipe lineage [final annotation]") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_full_Final_annotation.txt', checkIfExists: true) ]
                ]

                input[1] = [ file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/lineage_markers.txt', checkIfExists: true) ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("varpipe lineage [final annotation] -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file('test_full_Final_annotation.txt') ]
                ]

                input[1] = [ file('lineage_markers.txt') ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
