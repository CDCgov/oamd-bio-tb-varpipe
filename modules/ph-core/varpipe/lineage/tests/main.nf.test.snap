{
    "varpipe lineage [final annotation] -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_Lineage.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.lineage_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,dd0bac08142645010b31f3c81e4664eb"
                ],
                "lineage": [
                    [
                        {
                            "id": "test"
                        },
                        "test_Lineage.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "lineage_report": [
                    [
                        {
                            "id": "test"
                        },
                        "test.lineage_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dd0bac08142645010b31f3c81e4664eb"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-25T16:47:44.847486587"
    },
    "varpipe lineage [final annotation]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_Lineage.txt:md5,7e8e1b315c0e99a5780b2ffc4cf352e9"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.lineage_report.txt:md5,8bf9b428059472d1b760c88e6fafd600"
                    ]
                ],
                "2": [
                    "versions.yml:md5,eadb57f3e3381bea308fcf28e667089d"
                ],
                "lineage": [
                    [
                        {
                            "id": "test"
                        },
                        "test_Lineage.txt:md5,7e8e1b315c0e99a5780b2ffc4cf352e9"
                    ]
                ],
                "lineage_report": [
                    [
                        {
                            "id": "test"
                        },
                        "test.lineage_report.txt:md5,8bf9b428059472d1b760c88e6fafd600"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,eadb57f3e3381bea308fcf28e667089d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T15:39:48.038100135"
    }
}
